src/modules/admin/components/Tokens/TokensMoveDialogForm.tsx

```diff
@@ -0,0 +1,320 @@
+import React, { useMemo, useEffect, useCallback } from 'react';
+import { FormikProps } from 'formik';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import { COLONY_ROLE_FUNDING } from '@colony/colony-js-client';
+import BigNumber from 'bn.js';
+import { useMappedState } from 'redux-react-hook';
+
+import { Address } from '~types/index';
+import { DomainType } from '~immutable/index';
+import { useDataFetcher, useUserDomainRoles, useSelector } from '~utils/hooks';
+import Button from '~core/Button';
+import DialogSection from '~core/Dialog/DialogSection';
+import { Select, Input, FormStatus } from '~core/Fields';
+import Heading from '~core/Heading';
+
+import { domainsFetcher } from '../../../dashboard/fetchers';
+import { tokenBalanceSelector } from '../../../dashboard/selectors';
+import { walletAddressSelector } from '../../../users/selectors';
+
+import styles from './TokensMoveDialog.css';
+import { FormValues } from './TokensMoveDialog';
+import EthUsd from '~core/EthUsd';
+import Numeral from '~core/Numeral';
+import { ZERO_ADDRESS } from '~utils/web3/constants';
+
+const MSG = defineMessages({
+  title: {
+    id: 'admin.Tokens.TokensMoveDialog.title',
+    defaultMessage: 'Move Funds',
+  },
+  from: {
+    id: 'admin.Tokens.TokensMoveDialog.from',
+    defaultMessage: 'From',
+  },
+  to: {
+    id: 'admin.Tokens.TokensMoveDialog.to',
+    defaultMessage: 'To',
+  },
+  amount: {
+    id: 'admin.Tokens.TokensMoveDialog.amount',
+    defaultMessage: 'Amount',
+  },
+  domainTokenAmount: {
+    id: 'admin.Tokens.TokensMoveDialog.domainTokenAmount',
+    defaultMessage: 'Amount: {amount} {symbol}',
+  },
+  noBalance: {
+    id: 'admin.Tokens.TokensMoveDialog.noBalance',
+    defaultMessage: 'Insufficient balance in from domain pot',
+  },
+  noPermissionFrom: {
+    id: 'admin.Tokens.TokensMoveDialog.noPermissionFrom',
+    defaultMessage: 'No permission in from domain',
+  },
+  noPermissionTo: {
+    id: 'admin.Tokens.TokensMoveDialog.noPermissionTo',
+    defaultMessage: 'No permission in to domain',
+  },
+  samePot: {
+    id: 'admin.Tokens.TokensMoveDialog.samePot',
+    defaultMessage: 'Cannot move to same domain pot',
+  },
+});
+
+interface Props {
+  cancel: () => void;
+  colonyAddress: Address;
+  colonyTokenRefs: any;
+  colonyTokens: any;
+}
+
+const TokensMoveDialogForm = ({
+  cancel,
+  colonyAddress,
+  colonyTokens,
+  colonyTokenRefs,
+  handleSubmit,
+  isSubmitting,
+  isValid,
+  setErrors,
+  status,
+  values,
+}: Props & FormikProps<FormValues>) => {
+  // Find the currently selected token
+  const [selectedTokenRef, selectedToken] = useMemo(
+    () => [
+      colonyTokenRefs.find(token => token.address === values.tokenAddress),
+      colonyTokens.find(token => token.address === values.tokenAddress),
+    ],
+    [colonyTokenRefs, colonyTokens, values.tokenAddress],
+  );
+
+  // Map the colony's tokens to Select options
```

Comment:
I don't know if this was spec'd or not, but shouldn't we sort the native token first ?

![Screenshot from 2019-09-25 09-19-45](https://user-images.githubusercontent.com/1193222/65575263-17cea900-df78-11e9-8715-3fe8dc048c63.png)